### DISCUSSION TITLE
초기 Sequence Number인 ISN을 0부터 시작하지 않고 난수를 생성해서 설정하는 이유가 무엇인가요?

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#72] [General]
[백준 - 2504. 괄호의 값](https://www.acmicpc.net/problem/2504)

**Comment by SionBackEnd** [comment]
두가지 이유가 존재합니다. 
1. 보안 : 예측 가능한 시퀀스 번호는 보안 문제를 야기할 수있기 때문에 0부터 4,204,967,295번 사이 숫자 중 하나가 적용됩니다. 
2. 다중 세션 간섭 : TCP와 Connection을 맺을 때 사용하는 포트는 유한 범위 내에서 사용하고 시간이 지남에 따라 재사용이 되는데, 이때 이미 사용된 포트번호를 다시 재사용하는 호스트가 순차적인 시퀀스 번호를 전송하게 된다면, 이전 Connection 호스트로부터 오는 패킷으로 인식할 수 있다. 따라서 시퀀스 시작 번호를 난수로 설정하게 된다. 

[참고1](https://networkengineering.stackexchange.com/questions/61234/why-the-seq-number-set-to-random-there-will-be-safer-in-tcp-connect)
[참고2](https://lactea.kr/entry/Network-%E2%80%93-tcp-%EC%97%B0%EA%B2%B0-%EB%B6%84%EC%84%9D)
